omegaconf/dictconfig.py

```diff
@@ -292,9 +292,7 @@ def _s_validate_and_normalize_key(self, key_type: Any, key: Any) -> DictKeyType:
             return key  # type: ignore
         elif issubclass(key_type, Enum):
             try:
-                ret = EnumNode.validate_and_convert_to_enum(
-                    key_type, key, allow_none=False
-                )
+                ret = EnumNode.validate_and_convert_to_enum(key_type, key)
                 assert ret is not None
                 return ret
```

Comment:
It should be possible to remove this `assert ret is not None` line if the return type of `validate_and_convert_to_enum
` is updated.

Reply:
Right, also done in b556a82